build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -157,7 +158,22 @@ private static void RemovePdbsFromDir(string directory)
             FS.RmFilesInDirRecursive(directory, "*.pdb");
         }
 
-        private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli dotnet, string rootOutputDirectory)
+        private static BuildTargetResult CompileCliSdkAndGenerateNuGetPackagesArchive(
+            BuildTargetContext c,
+            DotNetCli dotnet,
+            string rootOutputDirectory,
+            DotNetCli currentDotnet)
```

Comment:
Is there a specific reason we can't just use Stage1 to generate the NuGet Packages Archive?  They are the same - the only difference is who built it.